fv3core/decorators.py

```diff
@@ -169,6 +169,121 @@ def _mark_cuda_fields_written(self, fields: Mapping[str, Storage]):
                 fields[write_field]._set_device_modified()
 
 
+class LoopStencil:
+    def __init__(
+        self,
+        func: Callable[..., None],
+        origins: List[Index3D],
+        domains: List[Index3D],
+        stencil_config: Optional[StencilConfig] = None,
+        externals: Optional[Mapping[str, Any]] = None,
+    ):
+        assert len(origins) == len(domains), (
+            "Lists of origins and domains need to have the same length, you provided "
+            + str(len(origins))
+            + " origins and "
+            + str(len(domains))
+            + " domains"
+        )
+        if externals is None:
+            externals = {}
+        if stencil_config is None:
+            stencil_config = global_config.get_stencil_config()
+        self._frozen_stencils = []
+        for index in range(len(origins)):
+            origin = origins[index]
+            domain = domains[index]
```

Comment:
Zip is fun for cases like this.

```suggestion
        for origin, domain in zip(origins, domains):
```

Reply:
thanks for reminding me, I know zip well! Just gotten in the habit in this repo of writing as straightforward python code as we can. but zip is probably ok!